ckanext/geocat/harvester.py

```diff
@@ -55,8 +65,14 @@ def _set_config(self, config_str, harvest_source_id):
             self.config = {}
 
         self.config['user'] = self.config.get('user', HARVEST_USER)
-        self.config['delete_missing_datasets'] = \
-            self.config.get('delete_missing_datasets', False)
+        self.config['rights'] = self.config.get('rights', DEFAULT_TERMS_OF_USE)
+        if not validate_terms_of_use(self.config['rights']):
+            self.config['rights'] = DEFAULT_TERMS_OF_USE
+
+        if self.config['delete_missing_datasets'].lower() == 'true':
+            self.config['delete_missing_datasets'] = True
+        else:
+            self.config['delete_missing_datasets'] = False
```

Comment:
Again, the boolean value from the json should be correctly converted to a bool when the config is loaded.